tests/models/test_models.py

```diff
@@ -416,8 +417,8 @@ def test_multihead_head_creation(self):
             w.data_ptr() for group in optimizer.param_groups for w in group["params"]
         ]
 
-        assert w_ptr not in opt_params_ptrs  # head0 has been updated
-        assert b_ptr not in opt_params_ptrs  # head0 has been updated
+        # assert w_ptr not in opt_params_ptrs  # head0 has NOT been updated
+        # assert b_ptr not in opt_params_ptrs  # head0 has NOT been updated
```

Comment:
Are you sure? I think it's if head0 is changed by SGD when training on task 4, which should not happen.

Reply:
w_ptr contains the head0 data pointer after head0 has been adapted on task 0. This pointer should still be the one present in optimizer after adaptation is done on task 4, since it should not be touched by the task 4 adaptation (which modifies head 4 only, and not head 0).